### PR TITLE
Add WavLM- & Wav2Vec2ForAudioFrameClassification support 

### PR DESCRIPTION
**Pull request by D4ve-R**
This adds support for WavLM- & Wav2Vec2ForAudioFrameClassification models.  
The models can be used for speaker diarization tasks.  
  
"Official model" [`microsoft/wavlm-base-plus-sd`](https://huggingface.co/microsoft/wavlm-base-plus-sd)

- [x] Add `AutoModelForAudioFrameClassification`
- [x] Add `Wav2Vec2ForAudioFrameClassification`
- [x] Add `WavLMForAudioFrameClassification`

### CODE REVIEW

**Comment by D4ve-R** [comment]
Transformers docs: 
[Wav2Vec2ForAudioFrameClassification](https://huggingface.co/docs/transformers/model_doc/wav2vec2#transformers.Wav2Vec2ForAudioFrameClassification). 
[WavLMForAudioFrameClassification](https://huggingface.co/docs/transformers/model_doc/wavlm#transformers.WavLMForAudioFrameClassification).

**Comment by xenova** [comment]
Thanks for the PR! Could you maybe explain how the output of the audio frame classification model should be interpreted? The example code in the [model card](https://huggingface.co/microsoft/wavlm-base-plus-sd#speaker-diarization) produces a one-hot array of shape `(num_frames, num_speakers)`, but it would be nice to be able to turn that into a JSON output with timestamps.

**Comment by D4ve-R** [comment]
Yes of course! The output logits are of dimension [num_batches, num_frames, num_speakers].
For now I can't tell you more, but I'm doing experiments, because I had the same idea 😆.
Since there is no `speaker-diarization` task in transformers, how do you feel about implementing a new pipeline for that task in transformers.js? I think it would be really cool.

File: src/models.js [hunk 1]
```diff
@@ -4778,6 +4792,50 @@ export class WavLMForXVector extends WavLMPreTrainedModel {
...
+ * //     size: 1098
+ * //   }
+ * // }
+ * // labels is a one-hot array of shape (num_batches, num_frames, num_speakers)
+ * const labels = logits
+ *  .sigmoid()
+ *  .toList()
+ *  .map(frames => frames.map(
+ *      frame => frame.map(speaker => speaker > 0.5 ? 1 : 0);
+ *   ));
```
Comment (xenova) [inline src/models.js:4825]: This code has a few issues in it (e.g., `toList` should be `tolist`). Can you update it and verify it works? Thanks!

Comment (xenova) [inline src/models.js:4825]:
```suggestion
 * import { AutoProcessor, AutoModelForAudioFrameClassification, read_audio } from '@xenova/transformers';
 * 
 * // Read and preprocess audio
 * const processor = await AutoProcessor.from_pretrained('Xenova/wavlm-base-plus-sd');
 * const url = 'https://huggingface.co/datasets/Xenova/transformers.js-docs/resolve/main/jfk.wav';
 * const audio = await read_audio(url, 16000);
 * const inputs = await processor(audio);
 * 
 * // Run model with inputs
 * const model = await AutoModelForAudioFrameClassification.from_pretrained('Xenova/wavlm-base-plus-sd');
 * const { logits } = await model(inputs);
 * // {
 * //   logits: Tensor {
 * //     dims: [ 1, 549, 2 ],  // [batch_size, num_frames, num_speakers]
 * //     type: 'float32',
 * //     data: Float32Array(1098) [-3.5301010608673096, ...],
 * //     size: 1098
 * //   }
 * // }
 * 
 * const labels = logits[0].sigmoid().tolist().map(
 *     frames => frames.map(speaker => speaker > 0.5 ? 1 : 0)
 * );
 * console.log(labels); // labels is a one-hot array of shape (num_frames, num_speakers)
 * // [
 * //     [0, 0], [0, 0], [0, 0], [0, 0], [0, 0], [0, 0],
 * //     [0, 0], [0, 0], [0, 0], [0, 0], [0, 0], [0, 0],
 * //     [0, 0], [0, 1], [0, 1], [0, 1], [0, 1], [0, 1],
 * //     ...
 * // ]
```

**Comment by flatsiedatsie** [comment]
@D4ve-R Did you manage to get the model to classify into three different speakers? The current inplementation in Transformers.js seems only split into two speakers.

**Comment by D4ve-R** [comment]
@flatsiedatsie sorry can't remember.

Here is a working example, to run speaker diarization with whisper + pyannote in transformer.js

https://huggingface.co/spaces/Xenova/whisper-speaker-diarization/blob/main/whisper-speaker-diarization/src/worker.js

Hope it helps

**Comment by flatsiedatsie** [comment]
Thanks, that's very kind.

I'm currently testing an implementation where I recursively re-segment long segments.
